src/evaluator.ts

```diff
@@ -614,6 +616,11 @@ class Evaluator {
       if (evalStep.test.assert?.some((a) => a.type === 'select-best')) {
         rowsWithSelectBestAssertion.add(row.testIdx);
       }
+      for (const assert of evalStep.test.assert || []) {
+        if (assert.type) {
+          assertionTypes.add(assert.type);
+        }
+      }
```

Comment:
We used to loop through all the test cases at the end to get the assertions. We'll track as we go now.